src/databricks/labs/ucx/azure/resources.py

```diff
@@ -121,44 +117,142 @@ def inner() -> dict[str, str]:
 
         return _credentials
 
+    def get(self, path: str, api_version: str | None = None):
+        headers = {"Accept": "application/json"}
+        query = {}
+        if api_version is not None:
+            query = {"api-version": api_version}
+        return self.api_client.do("GET", path, query, headers)
+
+    def put(self, path: str, api_version: str | None = None, body: dict | None = None):
+        headers = {"Content-Type": "application/json"}
+        query: dict[str, str] = {}
+        if api_version is not None:
+            query = {"api-version": api_version}
+        if body is not None:
+            return self.api_client.do("PUT", path, query, headers, body)
+        return None
+
+    def post(self, path: str, body: dict | None = None):
+        headers = {"Content-Type": "application/json"}
+        query: dict[str, str] = {}
+        if body is not None:
+            return self.api_client.do("POST", path, query, headers, body)
+        return self.api_client.do("POST", path, query, headers)
+
+    def delete(self, path: str):
+        # this method is added only to be used in int test to delete the application once tests pass
+        headers = {"Content-Type": "application/json"}
+        query: dict[str, str] = {}
+        return self.api_client.do("DELETE", path, query, headers)
+
+    def token(self):
+        return self._token_source.token()
+
+
+class AzureResources:
+    def __init__(self, azure_mgmt: AzureAPIClient, azure_graph: AzureAPIClient, include_subscriptions=None):
+        if not include_subscriptions:
+            include_subscriptions = []
+        self._azure_mgmt = azure_mgmt
+        self._azure_graph = azure_graph
```

Comment:
```suggestion
        self._mgmt = azure_mgmt
        self._graph = azure_graph
```
this class has `Azure` in the name. why do we need to duplicate that in properties?